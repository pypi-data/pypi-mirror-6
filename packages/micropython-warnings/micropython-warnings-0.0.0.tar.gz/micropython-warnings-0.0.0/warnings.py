# MicroPython doesn't support warnings
