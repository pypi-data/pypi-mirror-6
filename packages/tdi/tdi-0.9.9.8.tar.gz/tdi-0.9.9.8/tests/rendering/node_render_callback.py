#!/usr/bin/env python
import warnings as _warnings
_warnings.resetwarnings()
_warnings.filterwarnings('error')

from tdi import html

template = html.from_string("""
<html>
<body>
<div tdi="some">
content <tdi tdi="-count" />
</div>
</body>
</html>
""".strip())

class Model(object):
    def render_some(self, node):
        def callback(node, *fixed):
            node['fixed'] = ' '.join(map(str, fixed))
        print repr(node.render(callback))
        print repr(node.render(callback, decode=False))
        print repr(node.render(callback, 1, 2, 'a'))
        print repr(node.render(callback, 3, 4, 'b', decode=False))

    def render_count(self, node):
        node.content = self.count = getattr(self, 'count', 0) + 1

print template.render_string(Model())
