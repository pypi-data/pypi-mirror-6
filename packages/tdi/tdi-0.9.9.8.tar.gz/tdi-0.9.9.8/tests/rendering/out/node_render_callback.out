u'<div fixed="">\ncontent 1\n</div>'
'<div fixed="">\ncontent 2\n</div>'
u'<div fixed="1 2 a">\ncontent 3\n</div>'
'<div fixed="3 4 b">\ncontent 4\n</div>'
<html>
<body>
<div>
content 5
</div>
</body>
</html>
